sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/UnusedAsyncResultCheck.kt

```diff
@@ -0,0 +1,22 @@
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.js.descriptorUtils.getJetTypeFqName
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.bindingContextUtil.isUsedAsStatement
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.AbstractCheck
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+@Rule(key = "S6315")
+class UnusedAsyncResultCheck: AbstractCheck() {
+
+    override fun visitCallExpression(expression: KtCallExpression, context: KotlinFileContext) {
+        val bindingContext = context.bindingContext
+        if (bindingContext.get(BindingContext.EXPRESSION_TYPE_INFO, expression)?.type?.getJetTypeFqName(false) != "kotlinx.coroutines.Deferred") return
+        if (expression.isUsedAsStatement(bindingContext)) {
```

Comment:
Effectively these 2 if-blocks are a `&&` - I think it could make sense to write it as one as well, to be a bit more explicit about it